types/node/v16/module.d.ts

```diff
@@ -195,8 +195,14 @@ declare module "module" {
         static Module: typeof Module;
         constructor(id: string, parent?: Module);
     }
+    type ImportMetaDOMCompat = typeof globalThis extends { onmessage: any } ? {
+            resolve(specifier: string): string;
```

Comment:
In the compat case, can you still define the form with the optional param returning a promise? Or maybe that's overkill.

Reply:
nope:

```
../../../../ts/built/local/lib.es5.d.ts:631:11 - error TS2430: Interface 'ImportMeta' incorrectly extends interface '{ resolve(specifier: string, parent?: string | URL | undefined): Promise<string>; }'.
  The types returned by 'resolve(...)' are incompatible between these types.
    Type 'string' is not assignable to type 'Promise<string>'.
```